.github/workflows/tests.yaml

```diff
@@ -93,10 +110,22 @@ jobs:
               # https://github.com/dask/distributed/issues/4514
               export DISABLE_IPV6=1
           fi
-
           source continuous_integration/scripts/set_ulimit.sh
-          pytest distributed -m "not avoid_ci" --runslow
+
+          pytest distributed -m "not avoid_ci" --runslow \
+            --reruns 10 --reruns-delay 10 \
```

Comment:
This concerns me a little bit. Consider an actually broken CI run which fails every test. That would let a single job run for 10hr (assuming one full test suite runs about an hour). At the very least we should then shorten the GH actions timeout to not block CI for the entire org if something like this happens, see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

I would much rather stick to the selective rerunning.

Are these reruns marked in the reports generated here or will we be entirely blind if this happens? (other than parsing logs, ofc)

Reply:
Default GH action timeout is 5h. I've now shortened it to 3h. Hoping it's enough for when Windows decides to be slow...

Reply:
> Are these reruns marked in the reports generated here or will we be entirely blind if this happens?

Alas, you have to parse the logs. It's the same as before; it just applies to all tests instead of an ever-increasing selection.